### ISSUE TITLE
False positive "error: assignment to const variable [E003]"

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#155]
https://gist.github.com/strager/2695885dca0fe45cd2d2fc2dace0a35b

quick-lint-js reports the following:
```
hello.js:399:17: error: assignment to const variable [E003]
hello.js:395:19: note: const variable declared here [E003]
hello.js:406:17: error: assignment to const variable [E003]
hello.js:395:19: note: const variable declared here [E003]
```

These errors make absolutely no sense.

**Comment by strager** [comment]
I misread the code. I confused the declared variable `changePercent` with the undeclared (but assigned) variable `checkPercent`. Diagnostics look as expected.

**Comment by strager** [comment]
No, diagnostics don't look as expected. xD I'm not thinking straight, apparently.

**Comment by strager** [comment]
Reduced test case:
```js
const a = () => {
  b = null;
};
```
```
skywalk-bug.js:2:3: error: assignment to const variable [E003]
skywalk-bug.js:1:7: note: const variable declared here [E003]
skywalk-bug.js:2:3: warning: use of undeclared variable: b [E057]
```

**Comment by strager** [comment]
Fixed in commits 2092bef2af6072adf29b8268e5c8f1abd4640bd6 and 1bb43f9b87aae77921f0b074ef7ead2fa95f9c96. quick-lint-js now reports the following (ignoring references to `document`):

```
skywalk-bug.js:399:17: warning: assignment to undeclared variable [E059]
skywalk-bug.js:400:21: warning: use of undeclared variable: checkPips [E057]
skywalk-bug.js:406:17: warning: assignment to undeclared variable [E059]
skywalk-bug.js:407:21: warning: use of undeclared variable: checkPercent [E057]
```
